agent/config/config.go

```diff
@@ -233,19 +243,14 @@ func EnvironmentConfig() Config {
 	disableMetrics := utils.ParseBool(os.Getenv("ECS_DISABLE_METRICS"), false)
 	dockerGraphPath := os.Getenv("ECS_DOCKER_GRAPHPATH")
 
-	reservedMemoryEnv := os.Getenv("ECS_RESERVED_MEMORY")
-	var reservedMemory64 uint64
-	var reservedMemory uint16
-	if reservedMemoryEnv == "" {
-		reservedMemory = 0
-	} else {
-		reservedMemory64, err = strconv.ParseUint(reservedMemoryEnv, 10, 16)
-		if err != nil {
-			log.Warn("Invalid format for \"ECS_RESERVED_MEMORY\" environment variable; expected unsigned integer.", "err", err)
-			reservedMemory = 0
-		} else {
-			reservedMemory = uint16(reservedMemory64)
-		}
+	reservedMemory := parseEnvVariableUint16("ECS_RESERVED_MEMORY")
+
+	taskCleanupWaitDuration := parseEnvVariableDuration("ECS_ENGINE_TASK_CLEANUP_WAIT_DURATION")
+	// If a value has been set for taskCleanupWaitDuration and the value is less than the minimum allowed cleanup duration,
+	// print a warning and override it
+	if taskCleanupWaitDuration < minimumTaskCleanupWaitDuration {
```

Comment:
This logic should really be in `NewConfig` so it catches errors in `FileConfig` as well.